File: client/src/Persona.ts
export enum Dominio {
    Studente = "studente",
    Docente = "docente",
}

export abstract class Persona {
    abstract dominio: Dominio

    constructor(public id: string, public nome: string, public cognome: string, public dataNascita: Date) { }

    get email() {
        return (`${this.nome.charAt(0).toLowerCase()}${this.cognome.toLowerCase()}@${this.dominio}.com`);
    }
    CalcolaEta(): number {
        let oggi = new Date();
        let eta = oggi.getDate() - this.dataNascita.getDate();

        if (oggi.getMonth() < this.dataNascita.getMonth() ||
            (oggi.getMonth() === this.dataNascita.getMonth() && oggi.getDate() < this.dataNascita.getDate())) {
            eta--;
        }
        return eta;
    }

    getDati(): string {
        return (`${this.nome}${this.cognome}\n Identificativo: ${this.id}\n Età: ${this.CalcolaEta}\n Email: ${this.email}\n Specializzazione: ${Dominio}`)
    }

}

File: server/TestFineQuadrimestre/Dto/Persona.cs
namespace TestFineQuadrimestre.Dto
{

    public class Persona
    {
        int contatoreId = 1;
        public Guid Id { get; set; }
        public string? Nome { get; set; }
        public string? Cognome { get; set; }
        public DateTime DataNascita { get; set; }
        public Dominio Dominio { get; set; }
        public string? Email { get; set; }


        //public Persona(Guid id,string nome, string cognome, DateTime dataNascita, string email)
        //{
        //    Id = id;
        //    Nome = nome;
        //    Cognome = cognome;
        //    DataNascita = dataNascita;
        //    Email = email;
        //}
    }


}
